src/devices/tuya.ts

```diff
@@ -2502,6 +2499,45 @@ const definitions: Definition[] = [
             ],
         },
     },
+    {
+        fingerprint: [
+            {modelID: 'TS0601', manufacturerName: '_TZE200_7eue9vhc'},
+            {modelID: 'TS0601', manufacturerName: '_TZE200_bv1jcqqu'},
+        ],
+        whiteLabel: [
+            tuya.whitelabel('Zemismart', 'ZM25RX-08/30', 'Tubular motor', ['_TZE200_bv1jcqqu', '_TZE200_7eue9vhc']),
+        ],
+        model: 'TS0601_cover_8',
+        vendor: 'TuYa',
+        description: 'Cover motor',
+        onEvent: tuya.onEvent(),
+        configure: tuya.configureMagicPacket,
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        options: [exposes.options.invert_cover()],
+        exposes: [
+            e.text('work_state', ea.STATE),
+            e.cover_position().setAccess('position', ea.STATE_SET),
+            e.battery(),
+            // it also has `UPPER`/`LOWER` command that can be used to move motor over set limit, but it's not clear why :)
+            e.enum('program', ea.SET, ['SET BOTTOM', 'SET UPPER', 'RESET'/* , 'UPPER', 'LOWER' */]).withDescription('Set the upper/bottom limit'),
+            e.enum('motor_direction', ea.STATE_SET, ['NORMAL', 'REVERSED']).withDescription('Motor direction'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'state', tuya.valueConverterBasic.lookup({'OPEN': tuya.enum(0), 'STOP': tuya.enum(1), 'CLOSE': tuya.enum(2)})],
+                [2, 'position', tuya.valueConverter.coverPosition],
```

Comment:
The datapoints look very similar to the [TS0601_cover_4](https://github.com/Koenkk/zigbee-herdsman-converters/blob/3faa4dbab09da692981ca7368e1a5adbb0668dd2/src/devices/tuya.ts#L2606), can you check that the missing datapoints like `set_upper_limit` also work for this device?

Reply:
only`position`, `motor_direction` are working. also `opening_mode` mode overlaps with `tilt`/`lift`

should we try create device specific tuyaDatapoints? Something like that:

in whiteLabel we can set additional tags to give an alias to group of devices
```
        whiteLabel: [
            tuya.whitelabel('TuYa', 'LY-108', 'Cover', ['_TZE200_cpbo62rn'], ['tagA']),
            tuya.whitelabel('Zemismart', 'ZM25RX-08/30', 'Tubular motor', ['_TZE200_bv1jcqqu', '_TZE200_7eue9vhc'], ['tagB']),
        ],
```

then in meta we can create separate attribute for tag specific datapoints
```
        meta: {
            tuyaDatapoints: [
                [1, 'state', tuya.valueConverterBasic.lookup({'CLOSE': tuya.enum(2), 'STOP': tuya.enum(1), 'OPEN': tuya.enum(0)})],
                [2, 'position', tuya.valueConverter.coverPositionInverted],
                [3, 'position', tuya.valueConverter.coverPositionInverted],
                [7, 'work_state', tuya.valueConverterBasic.lookup({'standby': tuya.enum(0), 'success': tuya.enum(1), 'learning': tuya.enum(2)})],
                [13, 'battery', tuya.valueConverter.raw],
            ],
            tuyaDatapointsTagSpecific: {
                "tagA": [
                    [4, 'opening_mode', tuya.valueConverterBasic.lookup({'tilt': tuya.enum(0), 'lift': tuya.enum(1)})],
                    [101, 'motor_direction', tuya.valueConverterBasic.lookup({'left': tuya.enum(0), 'right': tuya.enum(1)})],
                    [107, 'factory_reset', tuya.valueConverter.setLimit],
                    [102, 'set_upper_limit', tuya.valueConverterBasic.lookup({'start': tuya.enum(1), 'stop': tuya.enum(0)})],
                    [101, 'motor_direction', tuya.valueConverterBasic.lookup({'left': tuya.enum(0), 'right': tuya.enum(1)})],
                ],
                "tagB": [
                    [5, 'motor_direction', tuya.valueConverterBasic.lookup({'NORMAL': tuya.enum(0), 'REVERSED': tuya.enum(1)})],
                    [101, 'program', tuya.valueConverterBasic.lookup({'SET BOTTOM': tuya.enum(0), 'SET UPPER': tuya.enum(1), 'RESET': tuya.enum(4),
                                                                    'LOWER': tuya.enum(2), 'UPPER': tuya.enum(3)})],
                ]
            }
        },
```

Reply:
I don't think we should go that route, it makes the definitions harder to read, we are basically creating a definition inside a defintion.

Since the only whitelabel is zemismart, I moved it to `zemismart.tsx`. I've also added some more comments after that is solved we can merge this.